taskograph.Web/Controllers/TaskController.cs

```diff
@@ -71,12 +81,141 @@ public IActionResult AddEntry(int taskId, long minutes)
 
         public IActionResult ConfigTasks()
         {
-            string _userId = User.FindFirstValue(ClaimTypes.NameIdentifier);
+            string _userId = GetIdentityUserId();
             ConfigTasksViewModel configTasksVM = new ConfigTasksViewModel();
             configTasksVM.Tasks = _taskRepository.GetAllTaskDTOs(_userId).ToList();
             configTasksVM.Groups = _groupRepository.GetAll(_userId).ToList();
 
             return View("ConfigTasks", configTasksVM);
         }
+
+        public IActionResult AddTask()
+        {
+            string _userId = GetIdentityUserId();
+            TaskViewModel taskVM = new TaskViewModel();
+            ReadGroupsSelectedItems(taskVM);
+            ReadColorsSelectedItems(taskVM);
+            taskVM.IsFormForTask = true;
+            return View("AddTask", taskVM);
+        }
+
+       
+
+        [HttpPost]
+        public IActionResult AddTask(TaskViewModel taskVM)
+        {
+            string _userId = GetIdentityUserId();
+            Task task = new Task()
+            {
+                Name = taskVM.Name,
+                GroupId = (taskVM.GroupId == UNASSIGNED_INT ? null : taskVM.GroupId),
+                ColorId = taskVM.ColorId,
+                AppUserId = _appUserRepository.GetId(_userId)
+            };
+            _taskRepository.Add(task);
+            return ConfigTasks();
+        }
+
+        public IActionResult AddGroup()
+        {
+            string _userId = GetIdentityUserId();
+            TaskViewModel taskVM = new TaskViewModel();
+            ReadColorsSelectedItems(taskVM);
+            ReadTasksSelectedItems(taskVM);
+            taskVM.IsFormForTask = false;
+            return View("AddTask", taskVM);
+        }
+
+        [HttpPost]
+        public IActionResult AddGroup(TaskViewModel taskVM)
+        {
+            string _userId = GetIdentityUserId();
+
+            int tempUserId = _appUserRepository.GetId(_userId);
+
+            Group group = new Group()
+            {
+                Name = taskVM.Name,
+                ColorId = taskVM.ColorId,
+                AppUserId = _appUserRepository.GetId(_userId)
+            };
+            _groupRepository.Add(group);
+            if (taskVM.TaskId != null && taskVM.TaskId != UNASSIGNED_INT)
+            {
+                Task task = _taskRepository.Get((int)taskVM.TaskId);
+                task.GroupId = group.Id;
+                _taskRepository.Edit(task);
```

Comment:
Skoro Twoja funkcja ma w sobie zarówno add jak i edit, to czy nie łamie single responsibility?

Reply:
Done